core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -187,6 +192,11 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
         stopInProgress = true
         context.become(stopping)
       }
+
+    case RequestMetrics =>
+      val unmodifiableYetMutableMetrics: java.util.Map[MetricName, _ <: Metric] = consumer.metrics()
```

Comment:
This map goes through a lot of conversions for the Java case.

I guess there's no point trying to avoid this. Looking at the Kafka code they give us the raw map wrapped in an unmodifiable so we shouldn't pass that around.

Reply:
Yeah it is converted a lot indeed... I figured to leave it along until reported as a problem